File: fluentcore/src/main/java/com/mcreater/fxfluent/util/ControlUtil.java
package com.mcreater.fxfluent.util;

import javafx.scene.Node;
import javafx.scene.control.SkinBase;
import javafx.scene.layout.Region;

public class ControlUtil {
    public static Region findControlInSkin(SkinBase<?> skin, String s) {
        final Node[] r = {null};
        skin.getChildren().forEach(node -> {
            if (node.getStyleClass().contains(s)) {
                r[0] = node;
            }
        });
        return (Region) r[0];
    }
}

File: fluentcore/src/main/java/com/mcreater/fxfluent/util/ControlUtil.kt
package com.mcreater.fxfluent.util

import javafx.geometry.Point2D
import javafx.scene.Node
import javafx.scene.control.SkinBase
import javafx.scene.layout.Region

class ControlUtil {
    companion object {
        @JvmStatic
        fun findControlInSkin(skin: SkinBase<*>, s: String?): Region? {
            val r = arrayOf<Node?>(null)
            skin.children.forEach {
                if (it.styleClass.contains(s)) r[0] = it
            }
            return r[0] as Region?
        }
        @JvmStatic
        fun displayPos(node: Node): Point2D {
            return node.localToScene(0.0, 0.0).add(Point2D(node.scene.window.x, node.scene.window.y))
        }
    }
}